### PR TITLE
[Document Repository] Fixed errors, warnings and added missing filter

### PR DESCRIPTION
**Pull request by AnyhowStep**
Category filter was not implemented. Implemented it, taking into account the tree structure categories have.

Adding a category was not working because an earlier merged pull request broke `NDB_Notifier_Abstract`. Changed `getCenterID()` to `getCenterIDs()`, to fix it.

Warnings were generated when adding a category, regarding uninitialized variables and calling `NDB_Client::initialize()` when `session_start()` has already been called, elsewhere (`session_start() already called, ignoring`).

Fixed warnings on `GET` of `/document_repository/`.

### CODE REVIEW

File: php/libraries/NDB_Notifier_Abstract.class.inc [hunk 1]
```diff
@@ -236,7 +236,7 @@ abstract class NDB_Notifier_Abstract
         $user =& User::singleton();
 
         //array of sites concerned
-        $sites_concerned =  $user->getCenterID();
+        $sites_concerned =  $user->getCenterIDs();
```
Comment (ridz1208) [inline php/libraries/NDB_Notifier_Abstract.class.inc]: @AnyhowStep will collide with https://github.com/aces/Loris/pull/2837/files

File: modules/document_repository/php/NDB_Menu_Filter_document_repository.class.inc [hunk 2]
```diff
@@ -77,9 +77,11 @@ class NDB_Menu_Filter_Document_Repository extends NDB_Menu_Filter
                           'v.Date_uploaded',
                          );
 
-        $openAccordion = $_GET['openAccordion'];
+        $openAccordion = isset($_GET['openAccordion']) ?
```
Comment (jstirling91) [inline modules/document_repository/php/NDB_Menu_Filter_document_repository.class.inc]: based on other pull request these should be switched to null operators

File: php/libraries/NDB_Notifier_Abstract.class.inc (hunk 1 above)
Comment (AnyhowStep) [inline php/libraries/NDB_Notifier_Abstract.class.inc]: The changes look the same, so, wouldn't the merge just work automagically?

File: modules/document_repository/php/NDB_Menu_Filter_document_repository.class.inc (hunk 2 above)
Comment (AnyhowStep) [inline modules/document_repository/php/NDB_Menu_Filter_document_repository.class.inc]: But this will still compile in php 7. So, it can still be merged now and will work.

Comment (AnyhowStep) [inline modules/document_repository/php/NDB_Menu_Filter_document_repository.class.inc]: Well, this is a pretty important fix, imo. So, I'll make the change now because I'm all for adopting php7.

We are moving to php7 before this release, right? Because this pull request needs to be in the release.

**Comment by AnyhowStep** [comment]
This will fail until our CI is changed to use php7 only.

**Comment by AnyhowStep** [comment]
This is no longer blocked.

**Comment by johnsaigle** [comment]
@ridz1208 @jstirling91 can you rereview?

File: php/libraries/NDB_Client.class.inc [hunk 3]
```diff
@@ -119,7 +119,9 @@ class NDB_Client
             . $config_additions
         );
         // start php session
-        session_start();
+        if (session_status() == PHP_SESSION_NONE) {
+            session_start();
+        }
```
Comment (jstirling91) [inline php/libraries/NDB_Client.class.inc]: @AnyhowStep How is this related to the pull request?

Comment (AnyhowStep) [inline php/libraries/NDB_Client.class.inc]: It was causing warnings.

http://php.net/manual/en/function.session-start.php
>4.3.3 	As of PHP 4.3.3, calling session_start() after the session was previously started will result in an error of level E_NOTICE. Also, the second session start will simply be ignored. 

If it's going to "simply be ignored", this pull request doesn't change existing business logic that is not causing failures while removing the E_NOTICE. This Document Repository module causes the warning to happen because it's calling `initialize()`

Comment (AnyhowStep) [inline php/libraries/NDB_Client.class.inc]: Particularly, here,
https://github.com/aces/Loris/pull/2820/files/ca9d8ccf14fc0087eee5c619b05f52f247396e3c#diff-a96dbaed094a9967897cbabb48ea169dR30

Comment (AnyhowStep) [inline php/libraries/NDB_Client.class.inc]: I already justified the code here,

>Warnings were generated when adding a category, regarding uninitialized variables and calling NDB_Client::initialize() when session_start() has already been called, elsewhere (session_start() already called, ignoring).

File: modules/document_repository/php/NDB_Menu_Filter_document_repository.class.inc [hunk 4]
```diff
@@ -77,9 +77,12 @@ class NDB_Menu_Filter_Document_Repository extends NDB_Menu_Filter
...
 
-        $openAccordion = $_GET['openAccordion'];
+        $openAccordion = $_GET['openAccordion'] ?? null;
+
         $this->tpl_data['openAccordion'] = $openAccordion;
-        $filtered = $_GET['filtered'];
+
+        $filtered = $_GET['filtered'] ?? null;
+
         $this->tpl_data['filtered'] = $filtered;
```
Comment (johnsaigle) [inline modules/document_repository/php/NDB_Menu_Filter_document_repository.class.inc]: This looks like a security issue. Please change to 
```php
$this->tpl_data['filtered'] = htmlspecialchars($filtered);
```

Comment (AnyhowStep) [inline modules/document_repository/php/NDB_Menu_Filter_document_repository.class.inc]: Yes, sir!

**Comment by johnsaigle** [comment]
@ridz1208 could you update your review?

**Comment by ridz1208** [comment]
@jstirling91 where are we on this ?

**Comment by kongtiaowang** [comment]
@AnyhowStep 
When I test this PR, It is missing the table header.
screen shot 2017-06-26 at 2 51 58 pm

**Comment by johnsaigle** [comment]
@AnyhowStep can you address the issue?

**Comment by johnsaigle** [comment]
Moving to 17.2 for now because this is unlikely to receive further development in the next while.

File: php/libraries/NDB_Client.class.inc (hunk 3 above)
Comment (ridz1208) [reply php/libraries/NDB_Client.class.inc]: @driusian someone should issue this seperately into the major branch and this should go to minor

**Comment by ridz1208** [comment]
Document repository will be part of a larger scale revamp. 

Redmine ticket 13569
